cmd/otp/otp_servehttp_test.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"bytes"
+	"crypto"
+	"github.com/go-logr/logr"
+	"github.com/go-logr/stdr"
+	"github.com/pkg/errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/rsa"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/crypto/ssh"
+)
+
+// A unit test for storekey.ServeHTTP and otp.ServeHTTP.
+// Tests happy and sad paths for each, each for their own specific functionality:
+//  1. otp.ServeHTTP checks it can provision a one time password properly when given an ssh key, and checks that if
+//     globalMap does not have a matching otp value, it returns the correct error and log writes
+//  2. storekey.ServeHTTP checks it can store a valid ssh key in globalMap, that it can store borderline-but-valid ssh
+//     keys in globalMap, and that it can validate an ssh key's structure to make sure it's indeed an ssh key and not
+//     some garbled/truncated/invalid content
+var _ = Describe("ServeHTTP handlers", Serial, func() {
+	var (
+		rr         *httptest.ResponseRecorder
+		rsaKey     string
+		req        *http.Request
+		err        error
+		logCapture *LogCapture
+		edKey      string
+	)
+
+	BeforeEach(func() {
+		rr = httptest.NewRecorder()
+		logCapture = NewLogCapture()
+		rsaKey, err = generateValidSSHKey("rsa")
+		Expect(err).ToNot(HaveOccurred())
+		edKey, err = generateValidSSHKey("ed25519")
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Describe("Testing otp.ServeHTTP", func() {
+		var testOtp *otp
+
+		JustBeforeEach(func() {
+			testOtp = NewOtp(&logCapture.Logger)
+		})
```

Comment:
why do we need this in a `JustBeforeEach`? Isn't `BeforeEach` enough?

```suggestion
		BeforeEach(func() {
			testOtp = NewOtp(&logCapture.Logger)
		})
```

Reply:
I thought, style-wise, that it is another way to compartmentalize the two test sessions for otp.ServeHTTP and storekey.HTTP. Also, saving memory and processing power, even in the slightest way. I spent two years developing a C++ real-time product, every bit mattered, and I guess the habit stuck.